### PR TITLE
Audio Playing

### PR DESCRIPTION
**Pull request by joyastockwell**

### CODE REVIEW

File: src/main/kotlin/app/ui/AudioPlayer.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import javafx.scene.media.AudioClip
+
+fun playKotlin(URI: String) {
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/AudioPlayer.kt]: kotlin is not a thing that is "played"

File: src/main/kotlin/app/ui/AudioPlayer.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    val myClip = AudioClip(URI)
+    myClip.play()
+    while(myClip.isPlaying) {
+        Thread.yield()
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/AudioPlayer.kt]: this is still a busy wait, and ultimately the player shouldn't be responsible for making this kind of thread management decision- AudioClip almost certainly spawns an internal thread to play on so what you're likely doing here is making decisions about the app's main thread.

File: src/main/kotlin/app/ui/AudioPlayer.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    }
+    println("done playing")
```
Comment (jsarabia) [inline src/main/kotlin/app/ui/AudioPlayer.kt]: We may want to introduce a logging class where things like this are done, but we probably don't want to be always writing things like this to standard out

**Comment by joyastockwell** [comment]
Ok. I will get on this when I am done with a first pass at the file creation methods for the user's audio hash and clip.

**Comment by joyastockwell** [comment]
Two questions:
1) I changed the function so it calls two helper functions: one to load from the URI and the other to play the loaded audio clip. Also, I want to put everything in a class so that I can mark that class as a singleton, but I don't know where I would put that class. Probably it shouldn't stay in the UI folder. Where's a good place for it?
2) Where should the threads get controlled from? Main? A thread controlling class? If the latter, should we have a separate branch for developing that?

**Pull request closed without merging**